tests/http/test_proxyagent.py

```diff
@@ -422,6 +423,133 @@ def test_https_request_via_proxy_with_blacklist(self):
         body = self.successResultOf(treq.content(resp))
         self.assertEqual(body, b"result")
 
+    def test_proxy_with_unsupported_scheme(self):
+        with self.assertRaises(ValueError):
+            _ = ProxyAgent(self.reactor, http_proxy=b"socks://proxy.com:8888")
+
+    def test_http_request_via_https_proxy(self):
+        agent = ProxyAgent(self.reactor, http_proxy=b"https://proxy.com:8888")
```

Comment:
It looks like this is missing the test certs.

```suggestion
        agent = ProxyAgent(self.reactor, contextFactory=get_test_https_policy(), http_proxy=b"https://proxy.com:8888")
```

See if this works.

Reply:
I've gone into a different direction with the test now, verifying that the proxy scheme parsing is working correctly and that the correct proxy_endpoint is constructed.

I think this is good enough as tests go, but feedback appreciated.